modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3947,7 +3947,9 @@ public CodegenProperty fromProperty(String name, Schema p, boolean required, boo
             List<Object> _enum = p.getEnum();
             property._enum = new ArrayList<>();
             for (Object i : _enum) {
-                property._enum.add(String.valueOf(i));
+                if(i != null) {
```

Comment:
I believe this can be rolled back, as `isValueNull` is sufficient.
The current model before this change is:

```
      "_enum" : [ "affinity-data", "dealroom", null ],
      "allowableValues" : {
        "enumVars" : [ {
          "name" : "AffinityData",
          "isValueNull" : false,
          "isString" : false,
          "value" : "'affinity-data'"
        }, {
          "name" : "Dealroom",
          "isValueNull" : false,
          "isString" : false,
          "value" : "'dealroom'"
        }, {
          "name" : "Null",
          "isValueNull" : true,
          "isString" : false,
          "value" : "'null'"
        } ],
        "values" : [ "affinity-data", "dealroom", null ]
      },
```

you can see there is an oddity in `allowableValues`, where the value is already serialized to `'null'`, however, #19027 didn't change this I believe, as it only touched `modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java` not the `DefaultCodegen`.

The `'null'` value is set here: https://github.com/joscha/openapi-generator/blob/85763cdb083a0803aba9fcbe88161326565667e1/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L6639

maybe the surrounding loop https://github.com/joscha/openapi-generator/blob/85763cdb083a0803aba9fcbe88161326565667e1/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L6626 needs the guard as well and we just remove the value from the possible enum values altogether? WDYT?

Reply:
@wing328 whats the core guidance here?

Reply:
I am just trying to guard the second part:

```diff
diff --git a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
index f27e4a400e4..9215daee503 100644
--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
@@ -6630,6 +6630,9 @@ public class DefaultCodegen implements CodegenConfig {
                 : 0;
 
         for (Object value : values) {
+            if(value == null) {
+                continue;
+            }
             Map<String, Object> enumVar = new HashMap<>();
             String enumName = truncateIdx == 0
                     ? String.valueOf(value)
```

and see how it will affect the samples. My guess is that it might be the better solution, as we don't introduce another template variable, keep the model more sane and it will fix all future generators as well.

Reply:
yes that would be nice, and it would work for all generators without modifying the templates.

Reply:
Okay, I will make the change later.

Just OOI: is there a better way to crate fixtures for a very specific part of the generation? It feels like quite the overkill to generate a whole scaffold with 10+ files just to assert one tiny part of it.
For the model change in this PR I could write a unit test, but generally, is there a way to create a partial scaffold in any way?

Reply:
just pushed 01209d53266 with the changes.

Reply:
looks good, thanks!

Reply:
> Just OOI: is there a better way to crate fixtures for a very specific part of the generation? It feels like quite the overkill to generate a whole scaffold with 10+ files just to assert one tiny part of it.
For the model change in this PR I could write a unit test, but generally, is there a way to create a partial scaffold in any way?

yeah unit tests are the more compact/preferred option for these specific aspects

Reply:
> yeah unit tests are the more compact/preferred option for these specific aspects

I can change this PR to use unit tests if/when @wing328 approves and we merge it.